packages/contract/src/contract-factory.ts

```diff
@@ -41,8 +41,8 @@ export type DeployContractOptions = {
 } & CreateTransactionRequestLike;
 
 export type DeployContractResult<TContract extends Contract = Contract> = {
-  transactionId: string;
   contractId: string;
+  getTransactionId: () => Promise<string>;
```

Comment:
Need some thoughts on this breaking change @FuelLabs/sdk-ts.

The problem with blob transactions is we don't have the transaction ID for the create TX until all blob TXs have been completed (because of spending / changing UTXOs that _can_ change when we come to funding the create TX.

To combat this transaction ID could become a promise. For `deployAsCreateTx` this can be returned as an immediately resolved promoise.

However for `deployAsBlobTx` this will be resolved once the blob transactions have completed and we have funded the create TX.

Reply:
Additional context [here](https://github.com/FuelLabs/fuels-ts/pull/2827/files#r1716619089).

Reply:
Or also open to other/fresh approaches to this problem.

Reply:
@danielbate I think your proposal to use promises is good enough.

However, if we add breaking changes, we must hurry up.

Reply:
I'll get it ready for review.

Reply:
I'm not entirely convinced this is a better approach than #2678 but let's see what people think.

Reply:
In case we don't have time for #2678 now, do you think we should still get this PR in and tackle that later?

Reply:
We definitely don't have time for #2678. The benefit of the type change is at least `string` -> `Promise<string>` gives us a lot more room to play with, regardless of the solution. I think it's worth getting this in.

Reply:
So is this problem a side effect of treating normal and big contract deployment as the same instead of letting the developers handle this on their own?

I do agree that `getTransactionId` breaking change seems to be the best approach for now.

Reply:
> So is this problem a side effect of treating normal and big contract deployment as the same instead of letting the developers handle this on their own?

Yes correct, in theory someone could manage the transactions themselves and all transaction IDs would be available in succession. But as the returned tx ID reflects the create tx ID, we don't have that till further in the flow for big deployments.

Reply:
> So is this problem a side effect of treating normal and big contract deployment as the same instead of letting the developers handle this on their own?

It does appear that way.

> I do agree that getTransactionId breaking change seems to be the best approach for now.

I agree with @Torres-ssf, it does seem the way forward now.

`waitForTxId` would be my preference of method name :)